splunk_topology/test_splunk_topology.py

```diff
@@ -79,8 +84,164 @@ def test_checks(self):
+        self.assertEquals(instances[0]["stop_snapshot"], True)
+
+        self.assertEquals(self.service_checks[0]['status'], 0, "service check should have status AgentCheck.OK")
+
+    def test_checks_backward_compatibility(self):
+        self.maxDiff = None
+
+        config = {
+            'init_config': {},
+            'instances': [
+                {
+                    'url': 'http://localhost:8089',
+                    'username': 'admin',
+                    'password': 'admin',
+                    'component_saved_searches': [{
+                        "name": "components",
+                        "parameters": {}
+                    }],
+                    'relation_saved_searches': [{
+                        "name": "relations",
+                        "parameters": {}
+                    }],
+                    'tags': ['mytag', 'mytag2']
+                }
+            ]
+        }
+
+        self.run_check(config, mocks={
+            '_dispatch_saved_search': _mocked_dispatch_saved_search,
+            '_search': _mocked_search,
+            '_saved_searches': _mocked_saved_searches,
+            '_auth_session': _mocked_auth_session
+        })
+
+        instances = self.check.get_topology_instances()
+        self.assertEqual(len(instances), 1)
+        self.assertEqual(instances[0]['instance'], {"type":"splunk","url":"http://localhost:8089"})
+
+        self.assertEqual(instances[0]['components'][0], {
+            "externalId": u"vm_2_1",
+            "type": {"name": u"vm"},
+            "data": {
+                u"running": True,
+                u"_time": u"2017-03-06T14:55:54.000+00:00",
+                "label.label1Key": "label1Value",
+                "tags": ['result_tag1', 'mytag', 'mytag2']
+            }
+        })
+
+        self.assertEqual(instances[0]['components'][1], {
+            "externalId": u"server_2",
+            "type": {"name": u"server"},
+            "data": {
+                u"description": u"My important server 2",
+                u"_time": u"2017-03-06T14:55:54.000+00:00",
+                "label.label2Key": "label2Value",
+                "tags": ['result_tag2', 'mytag', 'mytag2']
+            }
+        })
+
+        self.assertEquals(instances[0]['relations'][0], {
+            "externalId": u"vm_2_1-HOSTED_ON-server_2",
+            "type": {"name": u"HOSTED_ON"},
+            "sourceId": u"vm_2_1",
+            "targetId": u"server_2",
+            "data": {
+                u"description": u"Some relation",
+                u"_time": u"2017-03-06T15:10:57.000+00:00",
+                "tags": ['mytag', 'mytag2']
+            }
+        })
+
+        self.assertEquals(instances[0]["start_snapshot"], True)
+        self.assertEquals(instances[0]["stop_snapshot"], True)
+
+        self.assertEquals(self.service_checks[0]['status'], 0, "service check should have status AgentCheck.OK")
+
+    def test_checks_backward_compatibility_with_new_conf(self):
```

Comment:
I notice that this test is not in metric and not in event, do we need that there as well?

Reply:
@whazenberg  Same here. yeah just wanted to test if it works since all three check uses the same configuration instance from `splunk.py`. So thought to just test topology at least.